File: week6python/pset6/readability/readability.py
# coleman - Liau index
def main():
    textinput = input("Text: ")
    # calcualte nolof words, letters, sentences
    letters = count_letters(textinput)
    words = count_words(textinput)
    sentences = count_sentences(textinput)
    # calcualte the coleman-liau index
    index = indexCalculate(letters, words, sentences)
    #display index in required format
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")



def count_letters(inputline):
    letters = 0
    for i in range(len(inputline)):
        temp = ord(inputline[i])
        if (temp >= 65 and temp <= 90) or (temp >= 97 and temp <= 122):
            letters += 1
    return letters

def count_words(inputline):
    words = 0
    for i in range(len(inputline)):
        temp = inputline[i]
        if(temp == " "):
            words += 1
    return words+1

def count_sentences(inputline):
    sentences = 0
    for i in range(len(inputline)):
        temp = inputline[i]
        if temp == "." or temp == "?" or temp == "!":
            sentences += 1
    return sentences

def indexCalculate(letters, words, sentences):
    # calculate letters and sentences per 100 words 
    L = letters/words * 100
    S = sentences/words * 100
    # calcualre the index using the formula
    index = 0.0588*L - 0.296*S - 15.8
    # changing the index to an integer
    exactindex = round(index)

    return exactindex




if __name__ == "__main__":
    main()

File: week2arrays/pset2/readability/readability.c
// Coleman - Liau index
#include <string.h>
#include <stdio.h>
#include <math.h>

int count_letters(char *inputline);
int count_words(char *inputline);
int count_sentences(char *inputline);
int indexCalculate(int letters, int words, int sentences);

int main()
{
    // read the input line
    char textinput[999];
    printf("Text: ");
    fgets(textinput, 999, stdin);

    // calculate the no.of words , letters and sentences in line
    int letters = count_letters(textinput);
    int words = count_words(textinput);
    int sentences = count_sentences(textinput);

    // calculate the Coleman-Liau index
    int index = indexCalculate(letters, words, sentences);

    // display the index in required format
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }

    return 0;
}


int count_letters(char *inputline)
{
    int letters = 0;
    for (int i = 0, n = strlen(inputline); i < n; i += 1)
    {
        char temp = inputline[i];
        if ((temp >= 'a' && temp <= 'z') || (temp >= 'A' && temp <= 'Z'))
        {
            letters += 1;
        }
    }
    return letters;
}


int count_words(char *inputline)
{
    int words = 0;
    for (int i = 0, n = strlen(inputline); i < n; i += 1)
    {
        char temp = inputline[i];
        if (temp == ' ')
        {
            words += 1;
        }
    }

    return words + 1;
}


int count_sentences(char *inputline)
{
    int sentences = 0;
    for (int i = 0, n = strlen(inputline); i < n; i += 1)
    {
        char temp = inputline[i];
        if (temp == '.' || temp == '?' || temp == '!')
        {
            sentences += 1;
        }
    }

    return sentences;
}

int indexCalculate(int letters, int words, int sentences)
{
    // calculate the letters and sentences per 100 words
    float L = ((float)letters / words) * 100;
    float S = ((float)sentences / words) * 100;
    // calculate the index using the formula
    float index = 0.0588 * L - 0.296 * S - 15.8;
    // appropriately change the index to an integer
    int exactindex = (int)(round(index));

    return exactindex;
}
